### PR TITLE
 feat(scss): add _motion.scss

### PR DESCRIPTION
**Pull request by joshblack**
Partially addresses #1433 by including motion from elements, adding tests, and showing reference implementation in accordion. Work will need to be done to add to each component outside of this PR.

#### Changelog

**New**

- Add `_motion.scss` entrypoint with SassDoc for public motion variables, functions, and mixins
- Add test for `_motion.scss`

**Changed**

- Update `_vars.scss` to include `_motion.scss`

**Removed**

### CODE REVIEW

**Comment by joshblack** [comment]
cc @shixiedesign I updated the specs for accordion only, will follow up with the team to start thinking about how to update all the components listed in #1433 👍

File: src/components/accordion/_accordion.scss [hunk 1]
```diff
@@ -173,7 +173,7 @@
     position: relative;
     width: 100%;
     margin: 0;
-    transition: background-color $transition--base;
+    transition: background-color motion(standard, productive) 100ms;
```
Comment (tw15egan) [inline src/components/accordion/_accordion.scss:176]: Should we be using a variable here for the duration? (Goes for all the changes)

Comment (joshblack) [reply src/components/accordion/_accordion.scss:176]: @shixiedesign was saying that we don't currently have a token for these values so I wasn't sure 🤔

**Review by asudoh — APPROVED** [review]
LGTM except @tw15egan's comment - And thus once @shixiedesign signs off. Thanks @joshblack!

File: src/components/accordion/_accordion.scss (hunk 1 above)
Comment (tw15egan) [reply src/components/accordion/_accordion.scss:176]: @shixiedesign is there a reason why we don't have variables for these? If we don't, shouldn't we create them? I just want to ensure transitions are consistent throughout the library. Doesn't necessarily need to be implemented in this PR, but definitely something we should add

**Review by lovemecomputer — APPROVED** [review]
cool, exciting to get the motion!!

if we have common shared numbers for duration it would be cool to get variables for those, but i assume some components will have their own bespoke motion tuning too

File: src/components/accordion/_accordion.scss (hunk 1 above)
Comment (shixiedesign) [reply src/components/accordion/_accordion.scss:176]: right we should add variables. Currently we only have `$transition--base` and `$transition--expansion`, which is 100ms and 150ms. I wanna go in and add a few more. Can we wait for me to add a few more values, and then make a batch update to the duration values?

Comment (joshblack) [reply src/components/accordion/_accordion.scss:176]: Sure! That could work. When should I follow up for the new values?

**Review by shixiedesign — APPROVED** [review]
Yay! Can't wait to see all the motion implemented! 🎉

File: src/components/accordion/_accordion.scss (hunk 1 above)
Comment (shixiedesign) [reply src/components/accordion/_accordion.scss:176]: I'll make a PR for it soon!

Comment (shixiedesign) [reply src/components/accordion/_accordion.scss:176]: I meant to say update duration values to tokens above.

**Pull request merged**